packages/namekit-react/src/components/DisplayedPrice.tsx

```diff
@@ -0,0 +1,144 @@
+} as const;
+export type PriceDisplaySize =
+  (typeof PriceDisplaySize)[keyof typeof PriceDisplaySize];
+
+/**
+ * A map of the `PriceDisplaySize` to the `CurrencySymbolSize`.
+ * @param priceDisplaySize: PriceDisplaySize
+ * @returns CurrencySymbolSize
+ */
+export const parsePriceDisplaySizeToCurrencySymbolSize = (
+  priceDisplaySize: PriceDisplaySize,
+): CurrencySymbolSize => {
+  switch (priceDisplaySize) {
+    case PriceDisplaySize.Micro:
+    case PriceDisplaySize.Small:
+      return CurrencySymbolSize.Small;
+    case PriceDisplaySize.Medium:
+    case PriceDisplaySize.Large:
+      return CurrencySymbolSize.Large;
+  }
+};
+
+interface DisplayedPriceProps {
+  // The price to be displayed
+  price: Price;
+  // The onClick event handler for the price text
+  onTextClick?: () => void;
+  // Wether to display a currency symbology as its acronym or symbol
+  currencySymbology?: PriceSymbology;
+  // The place to display the currency symbology
+  symbolPosition?: CurrencySymbolPosition;
+  // The size of the price display
+  displaySize?: PriceDisplaySize;
+  // Wether or not to display the name of the currency in a tooltip when its symbol is hovered
+  describeCurrencyInTooltip?: boolean;
+  // Wether to display the currency symbology
+  withSymbol?: boolean;
+}
+
+export const DisplayedPrice = ({
+  price,
+  onTextClick,
+  describeCurrencyInTooltip = true,
+  symbolPosition = CurrencySymbolPosition.Left,
+  currencySymbology = PriceSymbology.Symbol,
+  displaySize = PriceDisplaySize.Small,
+  withSymbol = true,
+}: DisplayedPriceProps) => {
+  const priceDisplaySizeAsCurrencyDisplaySize =
+    parsePriceDisplaySizeToCurrencySymbolSize(displaySize);
+
+  return (
+    <div
+      role={onTextClick ? "button" : undefined}
+      onClick={onTextClick ? onTextClick : undefined}
+      className={cc([
+        "nk-min-w-max nk-inline-flex nk-justify-center nk-items-end nk-tabular-nums nk-leading-none",
+        displaySize,
+        {
+          "!nk-cursor-text": !onTextClick,
+          "!nk-cursor-pointer": onTextClick,
+        },
+      ])}
+    >
+      <p className="nk-order-2 nk-leading-none">{formattedPrice({ price })}</p>
+
+      <div
+        className={cc([
+          "nk-leading-none",
+          {
+            "nk-mr-1": price.currency === Currency.Usd,
+            "nk-order-1 nk-mr-1":
+              symbolPosition === CurrencySymbolPosition.Left,
+            "nk-mr-1 nk-order-3":
+              symbolPosition === CurrencySymbolPosition.Right,
+            "nk-hidden":
+              !withSymbol || currencySymbology === PriceSymbology.Acronym,
+          },
+        ])}
+      >
+        <CurrencySymbol
+          additionalClasses={onTextClick ? "nk-cursor-pointer" : ""}
+          describeCurrencyInTooltip={describeCurrencyInTooltip}
+          size={priceDisplaySizeAsCurrencyDisplaySize}
+          currency={price.currency}
+        />
+      </div>
+
+      {currencySymbology === PriceSymbology.Acronym && (
```

Comment:
Here, `DisplayedPrice` is handling the currency symbol rendering for the given price:

It is not possible to include this kind of logic in a children component since the father component, the wrapper, is the one responsible for positioning the currency symbol relatively to the price.

Reply:
It is pretty much saying:

Component A receives user input about **how** and **where** to place currency symbol relatively to price.
Component A renders either component B or C based on the **how**.
Component A always renders component D.

Component A then, is the one responsible for saying "Hey, based on user's preferences let's render either":
1. B + D
2. D + B
3. C + D
4. D + C

It is a wrapper that based on user preferences, displays the price with a currency symbology :)

I hope this clarifies what `DisplayedPrice` does, let me know if anything!